### DISCUSSION TITLE
Create Block Reference and copy embed link to clipboard

### DISCUSSION THREAD

**Discussion by GitMurf** [SilentVoid13/Templater#170] [Templates Showcase]
https://gist.github.com/GitMurf/c0500d8bcef2a0f83c0b526cf07fc265

QUICK DEMO: https://user-images.githubusercontent.com/64155612/115738150-a3214380-a341-11eb-9793-c874cb2ed2f6.mp4

Credit to: @shabegom - Add a block ref to the current line by adding a ^uniqueId on the end and copying to the clipboard the embed reference link so you can quickly paste the block ref somewhere else in your vault. Similar to Roam ctrl + click and drag to create block ref.

**Comment by welpdx** [comment]
This is really useful! Thanks

**Reply by welpdx** [reply]
I have an suggestion for an alternative script: prompt user for a manual block ID.

**Comment by classicrob** [comment]
Hey @GitMurf and @shabegom! Currently it's not working. Any advice? Is this an issue with the most recent version of Templater? I'm getting these errors in the console logs:

Uncaught (in promise) TypeError: this.templaterPlugin.templater.append_template is not a function
    at HotkeysForTemplates.eval (eval at <anonymous> (app.js:1), <anonymous>:2436:81)
    at step (eval at <anonymous> (app.js:1), <anonymous>:68:23)
    at Object.eval [as next] (eval at <anonymous> (app.js:1), <anonymous>:49:53)
    at eval (eval at <anonymous> (app.js:1), <anonymous>:42:71)
    at new Promise (<anonymous>)
    at __awaiter (eval at <anonymous> (app.js:1), <anonymous>:38:12)
    at HotkeysForTemplates.templaterInsertTemplate (eval at <anonymous> (app.js:1), <anonymous>:2427:16)
    at Object.callback (eval at <anonymous> (app.js:1), <anonymous>:2312:62)
    at vD (app.js:1)
    at e.executeCommandById (app.js:1)
Log.ts:17 Templater Error: Template parsing error, aborting. 
 Selection points outside of document
log_error @ Log.ts:17

**Reply by shabegom** [reply]
@classicrob

the issue is with this line:

`let curLine = cmEditorAct.getCursor().line;cmEditorAct.setSelection({ line: curLine, ch: 0 }, { line: curLine, ch: 9999 }); `

Specifically with the `ch: 9999` part. Change to this:

`let curLine = cmEditorAct.getCursor().line;cmEditorAct.setSelection({ line: curLine, ch: 0 }, { line: curLine, ch: Infinity }); `

Replacing `9999` with `Infinity` should make it work again.

**Reply by welpdx** [reply]
Sorry, I can't replicate your error. The script is working for me in builds public V0.12.19 and insider V0.13.5.

**Reply by classicrob** [reply]
> Replacing `9999` with `Infinity` should make it work again.

Thank you for the help! That worked.

**Comment by classicrob** [comment]
Hey! One more request - if there's already a block id at the end of the block, can this code be modified to copy the embed link to clipboard and skip the step of generating and adding a new id?

**Comment by GitMurf** [comment]
I don’t use this templater script anymore because Text Transporter plugin does this plus way more. Check it out.

https://github.com/TfTHacker/obsidian42-text-transporter#block-reference-commands
